### PR TITLE
api: Add missing GET teams endpoints

### LINKED ISSUE #1821: API : missing GET teams endpoint
Hi

Following #370 and #1809 , maybe miss at least to GET option 
* Looks to miss GET /api/v1/teams/:teamid/members/$USERNAME , should be return data about this username
* Looks to miss GET /api/v1/teams/ , should be return teams list related to the user query

And about admin usage (get all teams, ...) , should be set in /api/v1/admin/ part

### PR DESCRIPTION
**Pull request by HarshitOnGitHub**
This PR adds missing endpoints for teams as mentioned in issue #1821. I have not added endpoints for admin usage in this PR, will open a separate PR for it.
As this is my first PR to this project, I would appreciate a detailed review. Thanks! :)

Partly fixes: #1821.

### CODE REVIEW

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -610,6 +612,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 			m.Group("/members", func() {
 				m.Get("", org.GetTeamMembers)
 				m.Combo("/:username").
+					Get(org.GetTeamMember).
```
Comment (lunny) [inline routers/api/v1/api.go:657]: permission check.

File: routers/api/v1/org/team.go [hunk 2]
```diff
@@ -221,6 +244,35 @@ func GetTeamMembers(ctx *context.APIContext) {
...
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/User"
+	u := user.GetUserByParams(ctx)
+	if ctx.Written() {
+		return
+	}
+	ctx.JSON(200, u.APIFormat())
```
Comment (lunny) [inline routers/api/v1/org/team.go:286]: `u` maybe `nil`

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/api.go (hunk 1 above)
Comment (HarshitOnGitHub) [reply routers/api/v1/api.go:657]: What permission check are you suggesting to add? I think we are already checking for org membership?

File: routers/api/v1/org/team.go (hunk 2 above)
Comment (HarshitOnGitHub) [reply routers/api/v1/org/team.go:286]: Wouldn't in that case `ctx.Written()` condition will evaluate to true and function will exit early?

**Comment by HarshitOnGitHub** [comment]
@lunny Just a gentle ping in case you missed github notification for my reply. :)

**Comment by lunny** [comment]
@HarshitOnGitHub for compatible with github api v3, the route should be `/user/teams`, see https://developer.github.com/v3/teams/#list-user-teams, but currently you are in `/teams`? And the permission is right but I think maybe need an `organization` field on the return team list.

**Comment by HarshitOnGitHub** [comment]
@lunny Thanks for the review! I have updated the PR according to your comments, please take a look. :)

File: routers/api/v1/org/team.go [hunk 3]
```diff
@@ -42,6 +42,35 @@ func ListTeams(ctx *context.APIContext) {
...
+	//     "$ref": "#/responses/TeamList"
+	teams, err := models.GetUserTeams(ctx.User.ID)
+	if err != nil {
+		ctx.Error(500, "GetUserTeams", err)
+		return
+	}
+
+	apiTeams := make([]*api.Team, len(teams))
+	for i := range teams {
+		org, err := models.GetUserByID(teams[i].OrgID)
```
Comment (lunny) [inline routers/api/v1/org/team.go]: For better performance, maybe you could add a user cache via `var cache = make(map[int64]*User)`, if user are in many teams of one org that will reduce database queries.

Comment (HarshitOnGitHub) [reply routers/api/v1/org/team.go]: Agree! :)

**Comment by HarshitOnGitHub** [comment]
@lunny This is ready for a final review from you.

**Review by lunny — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**